lib/haskell/natural4/src/LS/XPile/MathLang/GenericMathLang/GenericMathLangAST.hs

```diff
@@ -44,11 +44,19 @@ import Language.Haskell.TH.Syntax (Lift)
 
 ------------ L4 declared entity types ----------------------
 
--- | Types that are declared in L4 by the user, e.g. 'Person' or 'Singaporean citizen'
-data L4EntType = L4EntType T.Text | L4Enum [T.Text] | L4List L4EntType
-  deriving stock (Show)
-  deriving (Eq, Generic, Hashable)
-
+-- | GML types that are declared in L4 by the user, e.g. 'Person' or 'Singaporean citizen'.
+--
+data L4EntType =
+    L4EntType T.Text  -- ^ a user-defined named type
+  | L4Enum [T.Text]   -- ^ a user-defined enumeration type
+  | L4List L4EntType  -- ^ a user-defined list type
+  deriving stock (Eq, Generic, Show)
+  deriving anyclass (Hashable)
+
+-- | Turn a surface-L4 type signature into a GML type.
+--
+-- NOTE / TODO: This translation seems lossy.
```

Comment:
You're right, it is lossy! I just never saw Optional or Set being used in any of the examples, and I'm not sure what the semantics of a list of an enum would be, so I left it simple (and) lossy like that.

Reply:
Sure, we should work on getting a better understanding what kind of stuff is used / not used over time. For now, I think it's ok to leave it and just add the note.